forgit.plugin.zsh

```diff
@@ -258,7 +258,17 @@ forgit::checkout::file() {
 # git checkout-branch selector
 forgit::checkout::branch() {
     forgit::inside_work_tree || return 1
-    [[ $# -ne 0 ]] && { git checkout -b "$@"; return $?; }
+    # if called with arguments, check if branch exists, else create a new one
+    if [[ $# -ne 0 ]]; then
+        if git show-branch "$@" &>/dev/null; then
+            git switch "$@"
+            return $?
+        else
+            git checkout -b "$@"
+            return $?
+        fi
+    fi
```

Comment:
If we can, it would be great to do similar to what we have in fish, that is in psuedocode:

```
saved_old_error_code = git switch_or_checkout
git status --short
return $checkout_status
```

Other than that this looks good! I'll add @wfxr to review, but with this requested change I'm happy to approve unilaterally, as I wrote the original fish PR

Reply:
Thanks for your feedback! Good idea, I have added the git status call.